dokuwiki-syntax/src/main/java/org/xwiki/contrib/dokuwiki/syntax/internal/parser/DokuWikiIterativeParser.java

```diff
@@ -858,9 +858,10 @@ private void processWord(StringBuilder word, Listener listener)
         word.setLength(0);
     }
 
-    private void processWordsFromReader(Reader source, Listener listener, String endString) throws IOException
+    private void processWordsFromReader(char ch, Reader source, Listener listener, String endString) throws IOException
     {
         ArrayList<Character> buffer = new ArrayList<>();
+        buffer.add(ch);
         int c = source.read();
```

Comment:
Again! It was the part of the current previous codebase. But I don't understand why c should be char and not int because with int we can check -1.

Reply:
You could cast it right after you checked for -1.

Reply:
> It was the part of the current previous codebase.

I comment the things I see :) Since you are doing important modifications to this method I suggest more improvements to this method.

Reply:
Thanks for the suggestion Thomas. :) 
I noticed after getting checked for -1. `int c` which is casted from `source.read()` is converted to `char` and then added to buffer.

Reply:
My point is actually that it's casted many times which is not very efficient.